### ISSUE TITLE
ExoPlayer PGSSUB implementation doesn't correctly render multiple on-screen captions 

### ISSUE THREAD

**Issue by silentTeee** [jellyfin/jellyfin-android#1045]
### Describe the bug

Unfortunately, there seems to be a major weakness in the ExoPlayer PGSSUB implementation: when more than one chunk of text is supposed to be on-screen at a time, (imagine one chunk for dialogue at the bottom, and another to caption a foreign billboard at the top), it will only display the second caption after removing the first. This can be extremely jarring when watching foreign films or shows, and depending on the timing for the text display, make following along impossible.

Here is a correct example of the rendering in ExoPlayer from 2.5.0-beta.6:
https://github.com/jellyfin/jellyfin-android/assets/20748060/e5cf3c23-1ee1-4e1b-9015-b152a295acdd

...and here is what happens when native PGSSUB rendering is enabled in ExoPlayer in the 2.5.0 official proprietary release: 
https://github.com/jellyfin/jellyfin-android/assets/20748060/9c5a44bc-60c0-496e-a3cd-078bb3c87205

 Keep in mind, this is a _very_ mild example. When a chunk of dialogue goes on for 10 seconds but some foreign on-screen text has a translation pop up just before the dialogue text does, a lot of information can be lost.

### Application version

2.5.0

### Where did you install the app from?

Sideloaded APK (proprietary build)

### Device information

Google Pixel 7

### Android version

Android 13

### Jellyfin server version

10.8.10

### Which video player implementations does this bug apply to?

- [ ] Web player (default)
- [X] Integrated player (ExoPlayer)
- [ ] External player (VLC, mpv, MX Player)

**Comment by nielsvanvelzen** [comment]
Probably this upstream issue google/ExoPlayer#7458

**Comment by silentTeee** [comment]
Yeah, based on the description that issue certainly sounds like what I was experiencing. Okay, so I can create a ticket in the AndroidTV repo sometime in the next 12 hours for this issue, since it happens on that app as well. 

As for a solution, would it be better to simply revert #1043 for now? It's been three years since the upstream issue was reported and it hasn't been resolved or shown any additional progress.

**Comment by silentTeee** [comment]
I suppose it could be hidden behind a configuration option, though IMO it should default to being disabled. 

I am admittedly a bit wary of keeping this feature for two reasons:

# Simply because I get an icky feeling whenever I release a feature that I am fully aware is buggy, _especially_ if I know there is no practical fix.
# In it's current state, how many people will such a feature actually benefit? 

My rationale for number 2 is this: so far, of all the discs I have seen produced in the USA, the commonly used subtitle format is PGS. This includes the discs of foreign productions. Some directors go out of their way to exclude shots of things with written text to avoid localization issues, but often they don't. 

And unless they are re-watching something, users won't know what's going to get captions and what won't, and as such can't predict what their experience will be. 

Ultimately, it's the decision of you lovely devs who are contributing your time on whether to keep this as an optional feature. I am merely sharing my opinion as a devoted user and fellow software developer.
